From: Stanislaw Kardach <stanislaw.kardach@caviumnetworks.com>
Date: Sun, 21 Feb 2016 16:28:02 +0100
Subject: [PATCH] Allow customizing Fuel commit

---
 build/config.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/config.mk b/build/config.mk
index 667575c..30d9ccc 100644
--- a/build/config.mk
+++ b/build/config.mk
@@ -9,7 +9,7 @@
 ##############################################################################
 
 FUEL_MAIN_REPO := https://github.com/openstack/fuel-main
-FUEL_MAIN_TAG = stable/8.0
+FUEL_MAIN_TAG := stable/8.0
 MOS_VERSION = 8.0
 OPENSTACK_VERSION = liberty-8.0
 
