From: Stanislaw Kardach <stanislaw.kardach@caviumnetworks.com>
Date: Fri, 4 Mar 2016 14:27:48 +0100
Subject: [PATCH] Allow customizing fuel-plugin-qemu

---
 build/f_isoroot/f_qemupluginbuild/config.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/f_isoroot/f_qemupluginbuild/config.mk b/build/f_isoroot/f_qemupluginbuild/config.mk
index 0b84654..e7fb65a 100644
--- a/build/f_isoroot/f_qemupluginbuild/config.mk
+++ b/build/f_isoroot/f_qemupluginbuild/config.mk
@@ -7,6 +7,6 @@
 # http://www.apache.org/licenses/LICENSE-2.0
 ##############################################################################
 
-QEMU_BRANCH=c35ce3377bcf382a4e36f1df49758100cd2910b1
-QEMU_REPO=https://review.openstack.org/openstack/fuel-plugin-qemu
+QEMU_BRANCH:=c35ce3377bcf382a4e36f1df49758100cd2910b1
+QEMU_REPO:=https://review.openstack.org/openstack/fuel-plugin-qemu
 QEMU_CHANGE=
