From: Florin Dumitrascu <florin.dumitrascu@enea.com>
Date: Thu, 17 Mar 2016 19:37:19 +0100
Subject: [PATCH] Allow customizing fuel-plugin-opendaylight

Signed-off-by: Florin Dumitrascu <florin.dumitrascu@enea.com>
---
 build/f_isoroot/f_odlpluginbuild/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/f_isoroot/f_odlpluginbuild/Makefile b/build/f_isoroot/f_odlpluginbuild/Makefile
index b0bc9c7..cc305e0 100644
--- a/build/f_isoroot/f_odlpluginbuild/Makefile
+++ b/build/f_isoroot/f_odlpluginbuild/Makefile
@@ -8,8 +8,8 @@
 ##############################################################################
 
 TOP := $(shell pwd)
-ODL_BRANCH=brahmaputra-release
-ODL_REPO="https://github.com/openstack/fuel-plugin-opendaylight.git"
+ODL_BRANCH:=brahmaputra-release
+ODL_REPO:="https://github.com/openstack/fuel-plugin-opendaylight.git"
 
 export USE_JAVA8=true
 export JAVA8_URL:=https://launchpad.net/~openjdk-r/+archive/ubuntu/ppa/+files/openjdk-8-jre-headless_8u72-b15-1~trusty1_amd64.deb
