From: Florin Dumitrascu <florin.dumitrascu@enea.com>
Date: Thu, 17 Mar 2016 19:37:19 +0100
Subject: [PATCH] Allow customizing fuel-plugin-opendaylight

Signed-off-by: Florin Dumitrascu <florin.dumitrascu@enea.com>
---
 build/f_isoroot/f_odlpluginbuild/config.mk | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/build/f_isoroot/f_odlpluginbuild/config.mk b/build/f_isoroot/f_odlpluginbuild/config.mk
index e64a05e..bca432c 100644
--- a/build/f_isoroot/f_odlpluginbuild/config.mk
+++ b/build/f_isoroot/f_odlpluginbuild/config.mk
@@ -7,9 +7,9 @@
 # http://www.apache.org/licenses/LICENSE-2.0
 ##############################################################################
 
-ODL_BRANCH=brahmaputra-sr2
-ODL_CHANGE=ec9474c1178e7451ec75f6e306ac84424a84a28a
-ODL_REPO="https://github.com/openstack/fuel-plugin-opendaylight.git"
+ODL_BRANCH:=brahmaputra-sr2
+ODL_CHANGE:=ec9474c1178e7451ec75f6e306ac84424a84a28a
+ODL_REPO:="https://github.com/openstack/fuel-plugin-opendaylight.git"
 
 export USE_JAVA8=true
 export JAVA8_URL:=https://launchpad.net/~openjdk-r/+archive/ubuntu/ppa/+files/openjdk-8-jre-headless_8u72-b15-1~trusty1_amd64.deb
