From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Thu, 18 Aug 2016 14:39:08 +0200
Subject: [PATCH] build: cache: Allow LOCAL_CACHE_ARCH_NAME override

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 ci/build.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ci/build.sh b/ci/build.sh
index 62abba1..82049cd 100755
--- a/ci/build.sh
+++ b/ci/build.sh
@@ -111,7 +111,7 @@ SCRIPT_DIR=$(readlink -f $(dirname ${BASH_SOURCE[0]}))
 BUILD_BASE=$(readlink -e ${SCRIPT_DIR}/../build/)
 RESULT_DIR="${BUILD_BASE}/release"
 BUILD_SPEC="${BUILD_BASE}/config.mk"
-LOCAL_CACHE_ARCH_NAME="fuel-cache"
+LOCAL_CACHE_ARCH_NAME="${LOCAL_CACHE_ARCH_NAME:-fuel-cache}"
 
 #
 # END of variables to customize
