diff --git a/build/.versions b/build/.versions
index e69de29..6595982 100644
--- a/build/.versions
+++ b/build/.versions
@@ -0,0 +1,5 @@
+fuel a365f05b903368225da3fea9aa42afc1d50dc9b4
+fuel 643a1ef27c7dccc1c2a2ad26b85c09226b35a67d
+fuel 643a1ef27c7dccc1c2a2ad26b85c09226b35a67d
+fuel 643a1ef27c7dccc1c2a2ad26b85c09226b35a67d
+fuel 643a1ef27c7dccc1c2a2ad26b85c09226b35a67d
diff --git a/build/config.mk b/build/config.mk
index e11d033..2a81027 100644
--- a/build/config.mk
+++ b/build/config.mk
@@ -9,7 +9,7 @@
 ##############################################################################
 
 FUEL_MAIN_REPO := https://github.com/openstack/fuel-main
-FUEL_MAIN_TAG = a365f05b903368225da3fea9aa42afc1d50dc9b4
+FUEL_MAIN_TAG :=
 MOS_VERSION = 8.0
 OPENSTACK_VERSION = liberty-8.0
 
diff --git a/build/docker/ubuntu-builder/Dockerfile b/build/docker/ubuntu-builder/Dockerfile
index e69de29..45fe992 100644
--- a/build/docker/ubuntu-builder/Dockerfile
+++ b/build/docker/ubuntu-builder/Dockerfile
@@ -0,0 +1,35 @@
+#!/bin/bash
+##############################################################################
+# Copyright (c) 2015 Ericsson AB and others.
+# stefan.k.berg@ericsson.com
+# jonas.bjurel@ericsson.com
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Apache License, Version 2.0
+# which accompanies this distribution, and is available at
+# http://www.apache.org/licenses/LICENSE-2.0
+##############################################################################
+
+FROM ubuntu:14.04
+
+RUN apt-get update
+RUN apt-get install -y software-properties-common python-software-properties \
+    make python-setuptools python-all dpkg-dev debhelper \
+    fuseiso git genisoimage bind9-host wget curl lintian tmux lxc iptables \
+    ca-certificates sudo apt-utils lsb-release dosfstools debmirror
+
+RUN echo "ALL ALL=NOPASSWD: ALL" > /etc/sudoers.d/open-sudo
+RUN echo "Defaults env_keep += \"ftp_proxy http_proxy https_proxy no_proxy RSYNC_PROXY RSYNC_CONNECT_PROG npm_config_registry\"" > /etc/sudoers.d/keep-proxies
+# Keeping PWD is needed to build as root
+RUN echo "Defaults env_keep += \"PWD\"" > /etc/sudoers.d/keep-pwd
+# Keeping variables for ISO build
+RUN echo "Defaults env_keep += \"MIRROR_UBUNTU MIRROR_UBUNTU_ROOT MIRROR_MOS_UBUNTU MIRROR_MOS_UBUNTU_ROOT MIRROR_FUEL LATEST_TARGET_UBUNTU\"" > /etc/sudoers.d/keep-mos
+RUN chmod 0440 /etc/sudoers.d/open-sudo
+RUN chmod 0440 /etc/sudoers.d/keep-proxies
+RUN chmod 0440 /etc/sudoers.d/keep-pwd
+RUN chmod 0440 /etc/sudoers.d/keep-mos
+RUN chmod 4755 /bin/fusermount
+
+ADD ./setcontext /root/setcontext
+RUN chmod +x /root/setcontext
+
+VOLUME /var/lib/docker
diff --git a/build/f_isoroot/f_repobuild/fuel_bootstrap_cli.yaml b/build/f_isoroot/f_repobuild/fuel_bootstrap_cli.yaml
index aa12c45..d630117 100644
--- a/build/f_isoroot/f_repobuild/fuel_bootstrap_cli.yaml
+++ b/build/f_isoroot/f_repobuild/fuel_bootstrap_cli.yaml
@@ -4,7 +4,7 @@
   extra_dirs:
     - /usr/share/fuel_bootstrap_cli/files/trusty
   output_dir: /tmp/
-  kernel_flavor: linux-image-generic-lts-trusty
+  kernel_flavor: linux-image-thunder
   packages:
     - vim
     - mc
@@ -54,7 +54,7 @@
       type: deb
     - name: mos
       section: "main restricted"
-      uri: "http://127.0.0.1:8080/ubuntu/x86_64"
+      uri: "http://127.0.0.1:8080/ubuntu/arm64"
       priority: 1050
       suite: mos8.0
       type: deb
