From: Stanislaw Kardach <stanislaw.kardach@caviumnetworks.com>
Date: Fri, 4 Mar 2016 14:26:43 +0100
Subject: [PATCH] Fix apt cache and qemu version handling

---
 qemu/build-qemu.sh | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/qemu/build-qemu.sh b/qemu/build-qemu.sh
index 652939e..f54f486 100755
--- a/qemu/build-qemu.sh
+++ b/qemu/build-qemu.sh
@@ -1,11 +1,12 @@
 #!/bin/bash
 wget http://wiki.qemu-project.org/download/qemu-2.2.1.tar.bz2
+sudo apt-get update -y
 sudo apt-get build-dep qemu -y
 sudo apt-get install devscripts -y
 sudo apt-get install dpkg-dev -y
-apt-get source qemu -y
-dpkg-source -x qemu_2.0.0+dfsg-2ubuntu1.21.dsc
-cd qemu-2.0.0+dfsg; uupdate -v 2.2.1 ../qemu-2.2.1.tar.bz2
+apt-get source qemu=2.0.0 -y
+dpkg-source -x qemu_2.0.0*.dsc
+cd qemu-2.0.0*; uupdate -v 2.2.1 ../qemu-2.2.1.tar.bz2
 cd ../qemu-2.2.1;echo "">> debian/patches/series
 sed -i 's/seccomp="yes"/seccomp="no"/' configure
 debian/rules build
-- 
1.9.1

